Support data auto-imports.

While normal functions are exported as "_foo" and "__imp__foo", data
symbols are exported as "__imp__foo" and "__nm__foo" in the implib
but are still "_foo" in the dependent code.

--- origsrc/flexdll/reloc.ml	2009-08-07 01:45:20.611397000 -0500
+++ src/flexdll/reloc.ml	2009-08-07 01:52:12.639945100 -0500
@@ -570,7 +570,9 @@ let build_dll link_exe output_file files
     List.iter
       (fun sym ->
 	 if Symbol.is_defin sym
-	 then add_def sym.sym_name
+	 then match check_prefix "__nm_" sym.sym_name with
+          | Some s -> add_def s;
+          | _ -> add_def sym.sym_name;
       )
       obj.symbols
   and collect_file fn =
@@ -717,8 +719,14 @@ let build_dll link_exe output_file files
               Printf.printf "%s -> %s(%s) because of %s\n%!" fn libname objname s;
             link_libobj o
           with Not_found ->
-            if !explain then
-              Printf.printf "%s needs %s (not found)\n%!" fn s
+              try
+              let (libname, objname, _) as o = Hashtbl.find defined_in ("__nm_" ^ s) in
+              if !explain then
+                Printf.printf "%s -> %s(%s) because of %s\n%!" fn libname objname s;
+              link_libobj o
+            with Not_found -> 
+              if !explain then
+                Printf.printf "%s needs %s (not found)\n%!" fn s
       )
       (needed obj)
   and link_libobj (libname,objname,obj) =
